.eslintrc.js

```diff
@@ -123,11 +67,96 @@ module.exports = {
       ),
     },
 
+    /**
+     * Files that require Apache 2.0 headers, settings
+     * are overridden below for files that require Elastic
+     * Licence headers
+     */
+    {
+      files: ['**/*.{js,ts,tsx}'],
+      rules: {
+        '@kbn/eslint/require-license-header': [
+          'error',
+          {
+            license: APACHE_2_0_LICENSE_HEADER,
+          },
+        ],
+        '@kbn/eslint/disallow-license-headers': [
+          'error',
+          {
+            licenses: [ELASTIC_LICENSE_HEADER],
+          },
+        ],
+      },
+    },
+
+    /**
+     * Files that require Elastic license headers instead of Apache 2.0 header
+     */
+    {
+      files: ['x-pack/**/*.{js,ts,tsx}'],
+      rules: {
+        '@kbn/eslint/require-license-header': [
+          'error',
+          {
+            license: ELASTIC_LICENSE_HEADER,
+          },
+        ],
+        '@kbn/eslint/disallow-license-headers': [
+          'error',
+          {
+            licenses: [APACHE_2_0_LICENSE_HEADER],
+          },
+        ],
+      },
+    },
+
+    /**
+     * Restricted paths
+     */
+    {
+      files: ['**/*.js'],
```

Comment:
Are we able to apply this to TS as well?

Reply:
I'll try it!